src/control/server/interceptors.go

```diff
@@ -133,3 +134,31 @@ func streamErrorInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.St
 	err := handler(srv, ss)
 	return proto.AnnotateError(err)
 }
+
+type statusGetter interface {
+	GetStatus() int32
+}
+
+// dErrFromStatus converts a numeric DAOS return status code
+// into an error.
+func dErrFromStatus(sg statusGetter) error {
+	dStatus := sg.GetStatus()
+	if dStatus == 0 {
+		return nil
+	}
+
+	return drpc.DaosStatus(-dStatus)
```

Comment:
The status coming back from the ioserver is already negative, isn't t?

Reply:
Oops, yeah. Fixed. For some reason I was thinking the response status was a uint32. Interestingly, d_errstr() gives the same results regardless of the sign. I'll add a quick test case for this helper, though.